### ISSUE TITLE
Allow for configuring a locale to be used for upper/lower casing inside of jOOQ

### ISSUE THREAD

**Issue by jazdw** [jOOQ/jOOQ#7153]
### Expected behavior and actual behavior:
Using RenderNameStyle.UPPER results in names being uppercased using the default Java locale. For example if the user has their locale set to Turkish then a table name of `dataPointTags` gets uppercased to `DATAPOİNTTAGS` (notice the dot above the I). This behaviour is explicitly mentioned in the [Java documentation](https://docs.oracle.com/javase/8/docs/api/java/lang/String.html#toUpperCase--).

I suspect the correct behaviour should be to always use `Locale.ROOT` when calling `String.toUpperCase()` and `String.toLowerCase()` although perhaps you would want to make this configurable from `Settings`.

This is the offending code from `DefaultRenderContext.literal(String literal)` although it looks like `RenderNameStyle` is used in a couple of other places too.
```
            if (LOWER == cachedRenderNameStyle)
                literal = literal.toLowerCase();
            else if (UPPER == cachedRenderNameStyle)
                literal = literal.toUpperCase();
```

### Steps to reproduce the problem:

1. Enable `configuration.settings().setRenderNameStyle(RenderNameStyle.UPPER)`
2. Set Java locale to Turkish
3. Try and use a table with an `i` in the table name

### Versions:

- jOOQ: 3.10.4
- Java: 8
- Database (include vendor): H2
- JDBC Driver (include name if inofficial driver): org.h2.Driver

**Comment by lukaseder** [comment]
Thanks for your report.

I don't think that jOOQ's behaviour is incorrect, after all, for the last 20 years, we've gotten used to Java's controversial design of applying default locales (and timezones, etc.) in the absence of explicit references.

Rather than changing the existing behaviour backwards incompatibly, I prefer adding a new `Setting` that allows for specifying the locale the should be used wherever applicable.

**Comment by jazdw** [comment]
>Thanks for your report.

No worries.

>I don't think that jOOQ's behaviour is incorrect, after all, for the last 20 years, we've gotten used to Java's controversial design of applying default locales (and timezones, etc.) in the absence of explicit references.

I would just think that for programmatic purposes it would make the most sense to use Locale.ROOT. At least for the scenario where I am using it, to uppercase names for use with H2. I assume that H2 does not do locale aware upper-casing.

>Rather than changing the existing behaviour backwards incompatibly, I prefer adding a new Setting that allows for specifying the locale the should be used wherever applicable.

Yep probably the safest option, thanks for adding it to the next milestone. I also noticed #7146, you might want to take this new setting into account there.

**Comment by lukaseder** [comment]
> I would just think that for programmatic purposes it would make the most sense to use Locale.ROOT

That would be a breaking change (probably), and I'm very reluctant to implement that.

> I also noticed #7146, you might want to take this new setting into account there.

Yes of course. It should apply everywhere.

**Comment by lukaseder** [comment]
Implemented as a new setting in jOOQ 3.12
